tests/snowpark/test_snowpark_shared.py

```diff
@@ -0,0 +1,137 @@
+
+import pytest
+import typer
+
+import snowcli.cli.snowpark_shared as shared
+from snowcli.utils import SplitRequirements
+from tests.testing_utils.fixtures import *
+
+
+@mock.patch("tests.snowpark.test_snowpark_shared.shared.utils.parse_anaconda_packages")
+@mock.patch("tests.snowpark.test_snowpark_shared.shared.utils.install_packages")
+def test_snowpark_package(
+    mock_install, mock_parse, temp_dir, correct_requirements_txt, caplog
+):
+
+    mock_parse.return_value = SplitRequirements(
+        [], [Requirement.parse("totally-awesome-package")]
+    )
+
+    mock_install.return_value = (True, None)
+
+    with caplog.at_level(logging.INFO, logger="snowcli.cli.snowpark_shared"):
+        result = shared.snowpark_package("yes", False, "yes")
+
+    assert "Comparing provided packages from Snowflake Anaconda..." in caplog.text
+
+    zip_path = os.path.join(temp_dir, "app.zip")
+    assert os.path.isfile(zip_path)
+
+    with ZipFile(zip_path) as zip:
+        assert "requirements.other.txt" in zip.namelist()
+
+        with zip.open("requirements.other.txt") as req_file:
+            reqs = req_file.readlines()
+            assert b"totally-awesome-package\n" in reqs
+
+
+@mock.patch("tests.snowpark.test_snowpark_shared.shared.utils.parse_anaconda_packages")
+@mock.patch("tests.snowpark.test_snowpark_shared.shared.utils.install_packages")
+def test_snowpark_package_with_packages_dir(
+    mock_install,
+    mock_parse,
+    temp_dir,
+    correct_requirements_txt,
+    dot_packages_directory,
+    caplog,
+):
+
+    mock_parse.return_value = SplitRequirements(
+        [], [Requirement.parse("totally-awesome-package")]
+    )
+    mock_install.return_value = (
+        True,
+        SplitRequirements([Requirement.parse("another-package-in-anaconda")], []),
+    )
+
+    with caplog.at_level(logging.INFO, logger="snowcli.cli.snowpark_shared"):
+        result = shared.snowpark_package("yes", False, "yes")
+
+    zip_path = os.path.join(temp_dir, "app.zip")
+    assert os.path.isfile(zip_path)
+
+    with ZipFile(zip_path) as zip:
+        assert "requirements.snowflake.txt" in zip.namelist()
+        assert (
+            os.path.join(
+                ".packages", "totally-awesome-package", "totally-awesome-module.py"
+            )
+            in zip.namelist()
+        )
+
+
+def test_snowpark_update_function_with_coverage_wrapper(caplog):
+
+    with caplog.at_level(logging.ERROR):
+        with pytest.raises(typer.Abort):
+            shared.snowpark_update(
+                type="function",
+                environment="dev",
+                name="hello",
+                file=Path("app.zip"),
+                handler="app.hello",
+                input_parameters="()",
+                return_type="str",
+                replace=False,
+                install_coverage_wrapper=True,
+            )
+
+    assert (
+        "You cannot install a code coverage wrapper on a function, only a procedure."
+        in caplog.text
+    )
```

Comment:
Instead of checking logs I would use mocks of connection, utils and check if proper functions are executed

Reply:
Done